pkg/action/programkind.go

```diff
@@ -78,22 +79,37 @@ func programKind(ctx context.Context, path string) string {
 
 	desc := ""
 	headerString := ""
+	var kind *magic.FileType
 	n, err := io.ReadFull(f, header[:])
 	if err == nil || errors.Is(err, io.ErrUnexpectedEOF) {
-		kind, err := magic.Lookup(header[:n])
-		if err == nil {
+		kind, err = magic.Lookup(header[:n])
+		if kind != nil && err == nil {
 			desc = kind.Description
 		}
 		headerString = string(header[:n])
 	}
 
-	// TODO: Is it safe to log unsanitized file stuff?
-	logger.Debug("magic", slog.String("desc", desc), slog.String("header", headerString), slog.Any("err", err))
-
 	if found, kind := byExtension(path); found {
+		// if we get a map hit, check if we're looking at an archive and that it has a common archive header
+		if _, ok := archiveMap[getExt(path)]; ok {
+			h := header[:n]
+			switch {
+			case bytes.HasPrefix(h, []byte("PK\x03\x04")) ||
```

Comment:
If we're going to do this kind of sniffing, we should import a better `magic` detection library instead of implementing it ourselves. I avoided doing so initially because they required cgo, but YARA does as well so it's no loss.

Reply:
Looks like `h2non/filetype` may be a good option: https://pkg.go.dev/github.com/h2non/filetype#IsArchive